### PR TITLE
HAAR-852: add restClient and set up manage-users-api to use it

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: backend/apiConfig.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import 'dotenv/config'
```
Comment (karenmillermoj) [inline backend/apiConfig.ts:1]: This file is part of config.ts in the template project. I have separated this out to make the file changes simpler.

File: backend/apis.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ const oauthApi = oauthApiFactory(
 const manageUsersApi = manageUsersApiFactory(
   clientFactory({
     baseUrl: config.apis.manageusers.url,
-    timeout: config.apis.manageusers.timeoutSeconds * 1000,
+    timeout: config.apis.manageusers.timeout.response,
```
Comment (karenmillermoj) [inline backend/apis.js:19]: Now just use the new one

File: backend/data/restClient.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
+import superagent from 'superagent'
+import Agent, { HttpsAgent } from 'agentkeepalive'
```
Comment (karenmillermoj) [inline backend/data/restClient.ts:2]: Copied from the template project - still need to add in metricsmiddleware

File: backend/sanitisedError.test.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import sanitisedError, { UnsanitisedError } from './sanitisedError'
```
Comment (karenmillermoj) [inline backend/sanitisedError.test.ts:1]: Copied from template project

File: backend/sanitisedError.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import type { ResponseError } from 'superagent'
```
Comment (karenmillermoj) [inline backend/sanitisedError.ts:1]: Copied from template project

File: backend/apiConfig.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+const production = process.env.NODE_ENV === 'production'
+type EnvOptions = { requireInProduction: boolean }
+const requiredInProduction: EnvOptions = { requireInProduction: true }
```
Comment (brightonsbox) [inline backend/apiConfig.ts:5]: Can this be removed? Only the notRequired... version appears to be used.

Comment (karenmillermoj) [reply backend/apiConfig.ts:5]: that is true - I left it in incase we need it - but YAGNI  - I'll remove

**Pull request merged**